benches/misc.rs

```diff
@@ -25,19 +24,21 @@ fn misc_gen_bool_const(b: &mut Bencher) {
 
 #[bench]
 fn misc_gen_bool_var(b: &mut Bencher) {
-    let mut rng = SmallRng::from_rng(&mut thread_rng()).unwrap();
+    let mut rng = StdRng::from_rng(&mut thread_rng()).unwrap();
     b.iter(|| {
+        let mut accum = true;
         let mut p = 0.18;
-        black_box(&mut p);  // Avoid constant folding.
         for _ in 0..::RAND_BENCH_N {
-            black_box(rng.gen_bool(p));
+            accum ^= rng.gen_bool(p);
+            p += 0.0001;
```

Comment:
Adjusting p seems sensible, but it adds an extra op, making this incomparable with `misc_gen_bool_const`. Was the old method not okay?

Reply:
Although LLVM didn't move the range set-up code out of the inner loop, I see no reason why it couldn't. Changing `p` makes sure it can not.

Funny thing: using `black_box` before the for loop is slower than incrementing `p` each round.